test/http/HttpRequest.test.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
Oh no, looks like your tests here are a lot longer than your actual code changes 🤯

Reply:
Yes there are some nuances to that rule

First, this PR deletes a net of 270 lines of test code (from 380 -> 110). I _could_ have kept basically all the tests from before, but I thought that would have been out of balance which is why I deleted the majority.

Second, when switching from one implementation to another, it's always good to keep some of the old tests to make sure you have a decent level of continuity.

Lastly, these tests fit into a category of integration testing where _our_ code isn't complex, but an underlying dependency has very complex code which is worth testing. You'll notice each test is a pretty basic scenario, and I haven't really left any tests for edge cases.